nixos/modules/services/web-apps/prosody-filer.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+
+  cfg = config.services.prosody-filer;
+
+  settingsFormat = pkgs.formats.toml { };
+  configFile = settingsFormat.generate "prosody-filer.toml" cfg.settings;
+in {
+
+  options = {
+    services.prosody-filer = {
+      enable = mkEnableOption "Prosody Filer XMPP upload file server";
+
+      settings = mkOption {
+        description = "Configuration for Prosody Filer.";
+
+        type = settingsFormat.type;
+
+        example = literalExample ''
+          {
+            secret = "mysecret";
+            storeDir = "/srv/http/nginx/prosody-upload";
+          }
+        '';
+
+        defaultText = literalExpression ''
+          {
+            listenport = mkDefault "127.0.0.1:5050";
+            uploadSubDir = mkDefault "upload/";
+          }
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.prosody-filer.settings = {
+      listenport = mkDefault "127.0.0.1:5050";
+      uploadSubDir = mkDefault "upload/";
```

Comment:
Where is this on the filesystem? What creates this directory?

Reply:
`uploadSubDir` is not a filesystem path, it's a sub-URL (`/upload` is the standard). There is also a required `storeDir` parameter, which can be situated in various places depending on your needs. Personally I have it pointing to `/srv/http/nginx/upload`, which I hook up to nginx (and create using `tmpfiles.d`).

Reply:
Ah perfect. Thanks!